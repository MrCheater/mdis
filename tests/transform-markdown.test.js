const path = require('path');

const { transformMarkdown } = require('../src/transform-markdown');

test('"transformMarkdown"', () => {
  const markdown = transformMarkdown(
    path.join(__dirname, 'files', 'markdown.md'),
    { encoding: 'utf8' }
  );

  expect(`${markdown}`).toMatchSnapshot();
});
