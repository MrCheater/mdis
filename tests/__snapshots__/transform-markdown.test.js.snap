// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"transformMarkdown" 1`] = `
"## named-blocks.js#imports

[mdis]:# (./named-blocks.js#imports)
\`\`\`
import React from 'react';
\`\`\`

## named-blocks.js#render

[mdis]:# (./named-blocks.js#render)
\`\`\`js
render() {
  return <div>Sample</div>;
}
\`\`\`

## simple

[mdis]:# (./simple.js)
\`\`\`js
class Point {
  ...
}
\`\`\`

## RAW file

[mdis]:# (./raw.js)
\`\`\`js
const store = new Map();

class Raw {
  get (key) {
    return store.get(key);
  }
  set (key, value) {
    return store.set(key, value);
  }
}

export default Raw
\`\`\`

End.

"
`;
